scipy/stats/_continuous_distns.py

```diff
@@ -2927,6 +2927,14 @@ def _entropy(self, c):
 gompertz = gompertz_gen(a=0.0, name='gompertz')
 
 
+def _average_with_log_weights(x, logweights):
```

Comment:
I verified this returns the same answers as the old code but without errors.

logweights - maxlogw is always <= 0 as mentioned so np.exp(...) never overflows :+1: